### ISSUE TITLE
unicorn/new-for-builtins disallows Object contructor.

### ISSUE THREAD

**Issue by ferhtgoldaraz** [sindresorhus/eslint-plugin-unicorn#917]
The Object constructor, when called without new, does a type conversion:

https://tc39.es/ecma262/#sec-object-constructor (section 19.1.1, it will call ToObject because NewTarget is the new object created when a construcor is called with new, and that is not there)

This type conversion, when an object is given, and only then, does nothing, preserving  the === identity of the given value.

This means that:

```js
const isObject = v => Object(v) === v;
```

...as a guard, guarantees that something can be used for method access (`foo.bar`).

This check is somehting I use a lot and has worked very well for me, I was suprised to find out it wouldn't be allowed.

unicorn/new-for-builtins

```js
const isObject = v => Object(v) === v;
```

**Comment by sindresorhus** [comment]
I guess we can just add an exception for exactly that. I don't want to remove `Object` as it's generally not something you should use.

**Comment by ferhtgoldaraz** [comment]
That would be enough I guess. I've only ever used it to compare for equality or inequality of some value with itself (`Object(value)` with `value`), and can't think of any ordinary thing I would ever do with it otherwise.
